ibis/expr/timecontext.py

```diff
@@ -127,6 +127,40 @@ def canonicalize_context(
     return begin, end
 
 
+def construct_multi_index_series(
+    series: pd.Series, frame: pd.DataFrame
+) -> pd.Series:
+    """ Construct a pd.MultiIndex of IntIndex and 'time'
+
+    In window execution, the result Series of udf may need
+    to be trimmed by timecontext. In order to do so, 'time'
+    must be added as an index to the Series. We extract
+    time column from the parent Dataframe `frame`.
+    See `trim_with_timecontext` in execution/window.py for
+    trimming implementation.
+
+    Parameters
+    ----------
+    series: pd.Series
+    frame: pd.DataFrame
+
+    Returns
```

Comment:
Can you add examples of shape of the index in various cases?

Reply:
Sure, different shapes of the index is unit tested in `test_timecontext.py`, but I also add actual examples in docstring for easier understading